### PR TITLE
mainnet test toggle background

### PR DESCRIPTION
**Pull request by piyalbasu**
This PR changes the network toggle from being a build time configuration to a run time configuration that defaults to mainnet

### CODE REVIEW

File: @shared/api/internal.ts [hunk 1]
```diff
@@ -279,7 +294,7 @@ export const retryAssetIcon = async ({
   } catch (e) {
     return assetIcons;
   }
-  const icon = await getIconUrlFromIssuer({ key, code });
+  const icon = await getIconUrlFromIssuer({ key, code, networkDetails });
```
Comment (piyalbasu) [inline @shared/api/internal.ts:297]: the only downside to not having a static network config is that in non-redux environments, we need to pass `networkDetails` as a param to any func that needs to access Horizon

Comment (piyalbasu) [reply @shared/api/internal.ts:297]: We could make the UI redux store a global, but it's a little messy to do so right now bc of monorepo structure. In the future, if passing around values becomes annoying, we can do some more work to do that

File: extension/src/background/messageListener/freighterApiMessageListener.ts [hunk 2]
```diff
@@ -74,9 +75,10 @@ export const freighterApiMessageListener = (
 
   const submitTransaction = async () => {
     const { transactionXdr } = request;
+    const { network, networkUrl } = getNetworkDetails(getIsTestnet());
```
Comment (piyalbasu) [inline extension/src/background/messageListener/freighterApiMessageListener.ts:78]: this is basically how it works now - figure out if we're on testnet based on localStorage, and then get the appropriate config

File: extension/src/popup/ducks/settings.ts [hunk 3]
```diff
@@ -55,26 +59,28 @@ const settingsSlice = createSlice({
...
-        const { isDataSharingAllowed } = action?.payload || {
-          isDataSharingAllowed: false,
+        const { isDataSharingAllowed, networkDetails } = action?.payload || {
+          ...initialState,
         };
 
         return {
           ...state,
           isDataSharingAllowed,
+          networkDetails,
```
Comment (piyalbasu) [inline extension/src/popup/ducks/settings.ts:83]: for UI, ask the Background for network config and then put it into redux

**Review by quietbits — APPROVED** [review]

**Pull request merged**